### PR TITLE
Stop osd test case

### PR DESCRIPTION
**Pull request by fbalak**
Add test_ceph_osd_stopped test case.

### CODE REVIEW

File: tests/manage/monitoring/conftest.py [hunk 1]
```diff
@@ -277,3 +277,59 @@ def stop_mon():
...
+
+
+@pytest.fixture
+def workload_stop_ceph_osd(measurement_dir):
+    """
+    Downscales Ceph osd deployment, measures the time when it was
+    downscaled and alerts that were triggered during this event.
+
+    Returns:
+        dict: Contains information about `start` and `stop` time for stopping
```
Comment (RazTamir) [inline tests/manage/monitoring/conftest.py]: Will be good to add example of that dict

File: tests/manage/monitoring/conftest.py [hunk 2]
```diff
@@ -277,3 +277,59 @@ def stop_mon():
         assert check_old_mons_deleted, msg
 
     return measured_op
+
+
+@pytest.fixture
+def workload_stop_ceph_osd(measurement_dir):
```
Comment (vasukulkarni) [inline tests/manage/monitoring/conftest.py]: name should be measure_ceph_stop_osd for better representing what it does.

File: tests/manage/monitoring/conftest.py [hunk 3]
```diff
@@ -277,3 +277,59 @@ def stop_mon():
...
+    osd_to_stop = osds[-1]
+    logger.info(f"osd disks to stop: {osd_to_stop}")
+    logger.info(f"osd disks left to run: {osds[:-1]}")
+
+    def stop_osd():
+        """
+        Downscale Ceph osd deployments for 11 minutes. First 1 minutes
+        the alert CephOSDDiskNotResponding should be in 'Pending'. After 1 minute
+        the alert turns into 'Firing' state.        This configuration of osd can be observed in ceph-mixins which
+        are used in the project:
```
Comment (vasukulkarni) [inline tests/manage/monitoring/conftest.py]: docstring needs formatting.

File: tests/manage/monitoring/conftest.py [hunk 4]
```diff
@@ -277,3 +277,59 @@ def stop_mon():
...
+            https://github.com/ceph/ceph-mixins/blob/d22afe8c0da34490cb77e52a202eefcf4f62a869/config.libsonnet#L21
+        There should be also CephClusterWarningState alert that takes 10
+        minutest to be firing.
+
+        Returns:
+            str: Names of downscaled deployments
+        """
+        # run_time of operation
+        run_time = 60 * 11
```
Comment (vasukulkarni) [inline tests/manage/monitoring/conftest.py]: could be parameterized?

File: tests/manage/monitoring/conftest.py [hunk 5]
```diff
@@ -277,3 +277,59 @@ def stop_mon():
...
+        dict: Contains information about `start` and `stop` time for stopping
+            Ceph osd pod
+    """
+    oc = ocp.OCP(
+        kind=constants.DEPLOYMENT,
+        namespace=config.ENV_DATA['cluster_namespace']
+    )
+    osd_deployments = oc.get(selector=constants.OSD_APP_LABEL)['items']
+    osds = [
+        deployment['metadata']['name']
```
Comment (ebenahar) [inline tests/manage/monitoring/conftest.py]: use .get()

Comment (fbalak) [reply tests/manage/monitoring/conftest.py]: done

File: tests/manage/monitoring/conftest.py (hunk 4 above)
Comment (fbalak) [reply tests/manage/monitoring/conftest.py]: I don't think that it should be parametrized as it is specific for the alert to be triggered. What json did you mean? If you mean ceph-mixins configuration then I wouldn't do that as it is external upstream project. Also if we don't load the value dynamically and if the test starts to fail in CI due to change in times for status change then we can easy notice that change and resolve it.

File: tests/manage/monitoring/conftest.py (hunk 2 above)
Comment (fbalak) [reply tests/manage/monitoring/conftest.py]: Good idea! I will change it next PR so it is changed for all alerting tests at once. Fixtures for all alerting tests are currently named this way but this PR is only about one test case.

**Pull request merged**